### PR TITLE
PeriodicResample Op: Fixes #9369 (clean history)

### LINKED ISSUE #9369: Sub-pixel shuffling tensor operation
Is there interest in integrating a [subpixel shuffling](http://www.cv-foundation.org/openaccess/content_cvpr_2016/papers/Shi_Real-Time_Single_Image_CVPR_2016_paper.pdf) operation as a core operation to Tensorflow? I've implemented such an operation on my [fork](https://github.com/jhetherly/tensorflow/tree/shuffle_op) and I think this could find widespread use. My implementation is a bit more generic than the one in the paper and is implemented using the C++ Tensorflow API. The operation I've defined is dimension agnostic and is found in the `contrib` portion of the source tree. Let me know if there is anything I need to do before making a pull request (i.e. better documentation, etc.).

### PR DESCRIPTION
**Pull request by jhetherly**
This is the same PR as #9376 with no CLA issues

### CODE REVIEW

**Comment by martinwicke** [comment]
Jenkins, test this please.

**Review by martinwicke — APPROVED** [review]
Carrying over review from previous PR

File: tensorflow/contrib/cmake/tf_core_kernels.cmake [hunk 1]
```diff
@@ -92,6 +92,8 @@ if(tensorflow_BUILD_CONTRIB_KERNELS)
       "${tensorflow_source_dir}/tensorflow/contrib/nccl/kernels/nccl_manager.cc"
       "${tensorflow_source_dir}/tensorflow/contrib/nccl/kernels/nccl_ops.cc"
       "${tensorflow_source_dir}/tensorflow/contrib/nccl/ops/nccl_ops.cc"
+      "${tensorflow_source_dir}/tensorflow/contrib/periodic_resample/kernels/periodic_resample_op.cc"
+      "${tensorflow_source_dir}/tensorflow/contrib/periodic_resample/ops/array_ops.cc"
```
Comment (mrry) [inline tensorflow/contrib/cmake/tf_core_kernels.cmake]: (I just commented about this on the closed PR...) Revert these two lines and the corresponding changes on L166–167. These are basically adding two elements to a list and removing them again :). Since you're building an `AddUserOps` library in `tf_python.cmake`, you don't need any changes in this file.

**Review by mrry — CHANGES_REQUESTED** [review]

File: tensorflow/contrib/cmake/tf_core_kernels.cmake (hunk 1 above)
Comment (jhetherly) [reply tensorflow/contrib/cmake/tf_core_kernels.cmake]: Fixed!

**Review by mrry — APPROVED** [review]
Thanks for persevering with this!

**Comment by mrry** [comment]
@tensorflow-jenkins test this please.

**Comment by martinwicke** [comment]
Jenkins, test this please. 

(I have merged fixes for these, so everything should pass)

File: tensorflow/contrib/cmake/tf_python.cmake [hunk 2]
```diff
@@ -224,7 +224,6 @@ add_python_module("tensorflow/python/grappler")
 add_python_module("tensorflow/python/keras")
 add_python_module("tensorflow/python/keras/activations")
 add_python_module("tensorflow/python/keras/applications")
-add_python_module("tensorflow/python/keras/applications/inception_resnet_v2")
```
Comment (martinwicke) [inline tensorflow/contrib/cmake/tf_python.cmake]: Bad merge? Deleting this line is causing the CMAKE failures.

**Review by martinwicke — CHANGES_REQUESTED** [review]

**Comment by martinwicke** [comment]
@jhetherly can you look into the test failures? I left a comment where I thought the cause is.

File: tensorflow/contrib/cmake/tf_python.cmake (hunk 2 above)
Comment (jhetherly) [reply tensorflow/contrib/cmake/tf_python.cmake]: added this line back in - have no idea how it was ever left out

**Comment by jhetherly** [comment]
Since I can't seem to find my previous comment I'll repeat it here; I've added back the Keras line that mysteriously got deleted in the CMake build. Should work now.

**Comment by mrry** [comment]
@tensorflow-jenkins test this please!

**Comment by jhetherly** [comment]
I'm not sure what is causing these failures, but it doesn't seem to have anything to do with my contribution. Could you rerun the tests?

**Comment by mrry** [comment]
@tensorflow-jenkins test this please.

**Review by martinwicke — APPROVED** [review]

**Comment by martinwicke** [comment]
There are some known unrelated failures, I'll look at it when the tests return.

**Comment by jhseu** [comment]
Jenkins, test this please

**Comment by jhetherly** [comment]
It seems like the issue was a Timeout on a test that appears unrelated to my contribution. Not sure how that could happen.

**Comment by martinwicke** [comment]
Yes, that's a known failure, you can ignore it.

**Comment by sb2nov** [comment]
@jhetherly can you look into the merge conflict

**Comment by jhetherly** [comment]
@sb2nov  fixed

**Comment by jhetherly** [comment]
@sb2nov I don't think these errors are due to any of my code.

**Comment by sb2nov** [comment]
I just fixed tests on master so will re-run these in a bit and then merge the PR. Sorry about the delay.

**Comment by sb2nov** [comment]
The build died midway, going to try one more time.
